cdci_data_analysis/pytest_fixtures.py

```diff
@@ -0,0 +1,251 @@
+                            pass
+                with open(self.output_file_path, 'w+') as outfile:
+                    l.append(obj_email_data)
+                    json.dump(l, outfile)
+
+            except Exception as e:
+                return '500 Could not process your message'
+            return '250 OK'
+
+    id = u''.join(random.choice(string.ascii_uppercase + string.digits) for _ in range(16))
+    if not os.path.exists('local_smtp_log'):
+        os.makedirs('local_smtp_log')
+    handler = CustomHandler(f'local_smtp_log/{id}_local_smtp_output.json')
+    controller = CustomController(id, handler, hostname='127.0.0.1', port=1025)
+    # Run the event loop in a separate thread
+    controller.start()
+
+    yield controller
+
+    print("will stop the mail server")
+    controller.stop()
+
+
+@pytest.fixture
+def dispatcher_local_mail_server_subprocess(pytestconfig):
+    import subprocess
+    import os
+    import copy
+    from threading import Thread
+
+    env = copy.deepcopy(dict(os.environ))
+    print(("rootdir", str(pytestconfig.rootdir)))
+    env['PYTHONPATH'] = str(pytestconfig.rootdir) + ":" + str(pytestconfig.rootdir) + "/tests:" + env.get('PYTHONPATH',
+                                                                                                          "")
+    print(("pythonpath", env['PYTHONPATH']))
+
+    cmd = [
+        "python",
+        " -m smtpd",
+        "-c DebuggingServer",
+        "-n localhost:1025"
+    ]
+
+    p = subprocess.Popen(
+        cmd,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        shell=False,
+        env=env,
+    )
+
+    def follow_output():
+        for line in iter(p.stdout):
+            line = line.decode()
+            print(f"mail server: {line.rstrip()}")
+
+    thread = Thread(target=follow_output, args=())
+    thread.start()
+
+    yield thread
+
+    print("will stop local mail server")
+    print(("child:", p.pid))
+    import os, signal
+    kill_child_processes(p.pid, signal.SIGKILL)
+    os.kill(p.pid, signal.SIGKILL)
+
+@pytest.fixture
+def dispatcher_test_conf(tmpdir):
+    fn = os.path.join(tmpdir, "test-dispatcher-conf.yaml")
+    with open(fn, "w") as f:
+        f.write("""
+dispatcher:
+    dummy_cache: dummy-cache
+    products_url: http://www.astro.unige.ch/cdci/astrooda_
+    dispatcher_url: 0.0.0.0
+    dispatcher_port: 8001
+    dispatcher_service_url: http://localhost:8001
+    sentry_url: "https://2ba7e5918358439485632251fa73658c@sentry.io/1467382"
+    logstash_host: 10.194.169.75
+    logstash_port: 5001
+    secret_key: 'secretkey_test'
+    email_options:
+        smtp_server: 'localhost'
+        sender_mail: 'team@odahub.io'
+        cc_receivers_mail: ['team@odahub.io']
+        smtp_port: 1025
+        smtp_server_password: ''
+        email_sending_timeout: True
+        email_sending_timeout_default_threshold: 1800
+        email_sending_job_submitted: True
+    """)
+
```

Comment:
Like this, don't you end up writing within the 'test-dispatcher-conf.yaml' file the same content?

Reply:
yes. but it's easier to do like this that to distribute and then discover yet another non-code file. see how discovery of the test*yaml was rather cumbersome

Reply:
Nice, it will just have to be aligned with the latest version of the configuration